internal/pkg/manifest/union.go

```diff
@@ -62,48 +66,64 @@ func (t Union[_, _]) IsAdvanced() bool {
 }
 
 // UnmarshalYAML decodes value into either type Basic or Advanced, and stores that value
-// in t. value is first decoded into type Basic, and t will hold type Basic if:
-//   - There was no error decoding value into type Basic
-//   - Basic.IsZero() returns false OR Basic does not support IsZero()
+// in t. value is first decoded into type Basic, and t will hold type Basic if
+// (1) There was no error decoding value into type Basic and
+// (2) Basic.IsZero() returns false OR Basic is not zero via reflection.
 //
 // If Basic didn't meet the above criteria, then value is decoded into type Advanced.
 // t will hold type Advanced if Advanced meets the same conditions that were required for type Basic.
 //
 // If value fails to decode into either type, or both types are zero after
 // decoding, t will not hold any value.
 func (t *Union[Basic, Advanced]) UnmarshalYAML(value *yaml.Node) error {
-	// reset struct
-	var simple Basic
-	var advanced Advanced
-	t.isBasic, t.Basic = false, simple
-	t.isAdvanced, t.Advanced = false, advanced
+	// convert value to []byte so we can use KnownFields()
+	// see https://github.com/go-yaml/yaml/issues/460
+	// side effect: TypeError line numbers are reset.
+	b, err := yaml.Marshal(value)
+	if err != nil {
+		return err
+	}
 
-	sErr := value.Decode(&simple)
-	if sErr == nil && !isZeroerAndZero(simple) {
-		t.isBasic, t.Basic = true, simple
+	var basic Basic
+	dec := yaml.NewDecoder(bytes.NewReader(b))
+	dec.KnownFields(true)
+	bErr := dec.Decode(&basic)
+	if bErr == nil && !isZero(basic) {
+		t.SetBasic(basic)
 		return nil
 	}
 
-	aErr := value.Decode(&advanced)
-	if aErr == nil && !isZeroerAndZero(advanced) {
-		t.isAdvanced, t.Advanced = true, advanced
+	var advanced Advanced
+	dec = yaml.NewDecoder(bytes.NewReader(b))
+	dec.KnownFields(true)
+	aErr := dec.Decode(&advanced)
+	if aErr == nil && !isZero(advanced) {
```

Comment:
I wonder if this works?
```suggestion
	if dec.Decode(&advanced) == nil && !isZero(advanced) {
```

Reply:
the error values are used later in the function!